pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -0,0 +1,330 @@
+	Spec *datadoghq.DatadogPodAutoscalerSpec
+
+	// SettingsTimestamp is the time when the settings were last updated
+	// Version is stored in .Spec.RemoteVersion
+	// (only if owner == remote)
+	SettingsTimestamp time.Time
+
+	// ScalingValues represents the current target scaling values (retrieved from RC)
+	ScalingValues ScalingValues
+
+	// ScalingValuesHash is the hash of the values from RC
+	ScalingValuesHash string
+
+	// ScalingValuesVersion is the version of the values from RC
+	ScalingValuesVersion *uint64
+
+	// ScalingValuesTimestamp is the time when the values were last updated
+	ScalingValuesTimestamp time.Time
+
+	// HorizontalLastAction is the last horizontal action successfully taken
+	HorizontalLastAction *datadoghq.DatadogPodAutoscalerHorizontalAction
+
+	// HorizontalLastActionError is the last error encountered on horizontal scaling
+	HorizontalLastActionError error
+
+	// VerticalLastAction is the last action taken by the Vertical Pod Autoscaler
+	VerticalLastAction *datadoghq.DatadogPodAutoscalerVerticalAction
+
+	// VerticalRolloutError is the last error encountered on vertical scaling
+	VerticalRolloutError error
+
+	// CurrentReplicas is the current number of PODs for the targetRef
+	CurrentReplicas *int32
+
+	// ScaledReplicas is the current number of PODs for the targetRef matching the resources recommendations
+	ScaledReplicas *int32
+
+	// GlobalError is the an error encountered by the controller not specific to a scaling action
+	GlobalError error
+
+	// Deleted flags the PodAutoscaler as deleted (removal to be handled by the controller)
+	// (only if owner == remote)
+	Deleted bool
+
+	//
+	// Private fields
+	//
+	// targetGVK is the GroupVersionKind of the target resource
+	// Parsed once from the .Spec.TargetRef
+	targetGVK schema.GroupVersionKind
+}
+
+// NewPodAutoscalerInternal creates a new PodAutoscalerInternal from a Kubernetes CR
+func NewPodAutoscalerInternal(podAutoscaler *datadoghq.DatadogPodAutoscaler) PodAutoscalerInternal {
+	pai := PodAutoscalerInternal{
+		Namespace: podAutoscaler.Namespace,
+		Name:      podAutoscaler.Name,
+	}
+	pai.UpdateFromPodAutoscaler(podAutoscaler)
+	pai.UpdateFromStatus(podAutoscaler)
+
+	return pai
+}
+
+// NewPodAutoscalerFromSettings creates a new PodAutoscalerInternal from settings received through remote configuration
+func NewPodAutoscalerFromSettings(ns, name string, podAutoscalerSpec *datadoghq.DatadogPodAutoscalerSpec, settingsVersion uint64, settingsTimestamp time.Time) PodAutoscalerInternal {
+	pda := PodAutoscalerInternal{
+		Namespace: ns,
+		Name:      name,
+	}
+	pda.UpdateFromSettings(podAutoscalerSpec, settingsVersion, settingsTimestamp)
+
+	return pda
+}
+
+// ID returns the functional identifier of the PodAutoscaler
+func (p *PodAutoscalerInternal) ID() string {
+	return p.Namespace + "/" + p.Name
+}
+
+// GetTargetGVK resolves the GroupVersionKind if empty and returns it
+func (p *PodAutoscalerInternal) GetTargetGVK() (schema.GroupVersionKind, error) {
```

Comment:
This function can return no error if both Group and Version are empty. Are we fine with that ?